mission_control/navigator_missions/navigator_singleton/navigator.py

```diff
@@ -221,6 +238,57 @@ def get_response(self, **kwargs):
 
         return self.client(s_req)
 
+class MissionParam(object):
+    def __init__(self, nh, param, options):
+        self.nh = nh
+        self.param = param
+        self.options = options
+
+    @util.cancellableInlineCallbacks
+    def get(self):
+        # Returns deferred object, make sure to yield on this (same for below)
+        if not (yield self.exists()):
+          raise Exception("Mission Param {} not yet set".format(self.param))
```

Comment:
Maybe have default values specified in the yaml that are loaded if a param isn't found?